src/auth/aws.js

```diff
@@ -45,6 +50,86 @@ export function createCanonicalRequest(request) {
     return `${CanonicalRequestObj.HTTPMethod}\n${CanonicalRequestObj.CanonicalURI}\n${CanonicalRequestObj.CanonicalQueryString}\n${CanonicalRequestObj.CanonicalHeaders}\n${CanonicalRequestObj.SignedHeaders}\n${CanonicalRequestObj.HashedPayload}\n`;
 }
 
-export function createStringToSign() {
 
+/**
+ * Returns string to sign
+ * 
+ * @export
+ * @param {any} canonicalRequest 
+ * @returns 
+ */
+export function createStringToSign(canonicalRequest) {
+    if (!canonicalRequest) throw Error(ERROR_PARAM_REQUIRED);
+    if (typeof canonicalRequest !== 'string') throw Error(ERROR_PARAM_TYPE_IS_NOT_STRING);
+
+    const timeStampISO8601Format = _.getAWSTimestamp();
+    const scope = _.createScope();
+    const request = hex(sha256(canonicalRequest).toString());
+
+    return `${AUTH_IDENTIFIER_HEADER}\n${timeStampISO8601Format}\n${scope}\n${request}`;
+}
+
+/**
+ * Returns Signing key
+ * 
+ * @export
+ * @param {any} awsSecretKey 
+ * @param {any} awsRegion 
+ * @returns {String}
+ */
+export function createSigningKey(awsSecretKey, awsRegion) {
+    if (!awsSecretKey) throw Error(ERROR_PARAM_REQUIRED);
+    if (typeof awsSecretKey !== 'string') throw Error(ERROR_PARAM_TYPE_IS_NOT_STRING);
+    if (!awsRegion) throw Error(ERROR_PARAM_REQUIRED);
+    if (typeof awsRegion !== 'string') throw Error(ERROR_PARAM_TYPE_IS_NOT_STRING);
```

Comment:
we could - but what's the disadvantage of doing it here? I do it for example with the request method...